contracts/lib.rs

```diff
@@ -123,28 +216,67 @@ mod open_payroll {
             self.paused_block_at.is_some()
         }
 
-        // ensure_in_not_paused ensures that the contract is not paused
-        fn ensure_in_not_paused(&self) -> Result<(), Error> {
+        // ensure_is_not_paused ensures that the contract is not paused
+        fn ensure_is_not_paused(&self) -> Result<(), Error> {
             if self.is_paused() {
                 return Err(Error::ContractIsPaused);
             }
             Ok(())
         }
 
+        fn check_multipliers_are_valid(
+            &self,
+            multipliers: &Vec<(MultiplierId, Multiplier)>,
+        ) -> Result<(), Error> {
+            for (multiplier_id, _) in multipliers.iter() {
+                if !self.base_multipliers.contains(multiplier_id) {
+                    return Err(Error::MultiplierNotFound);
+                }
+                if self
+                    .base_multipliers
+                    .get(multiplier_id)
+                    .unwrap()
+                    .deactivated_at
+                    .is_some()
+                {
+                    return Err(Error::MultiplierAlreadyDeactivated);
+                }
+            }
+            Ok(())
+        }
+
+        //TODO: Maybe this function could be generic over the type of the vec and the error type
+        // Can be used to check unique multipliers and also unique beneficiaries
+        fn check_no_duplicate_multipliers(
+            &self,
```

Comment:
This function should be pure? Not depend on self. Because just uses the parameter